### PR TITLE
Module#1

### PR DESCRIPTION
**Pull request by Fat-Frumos**

### CODE REVIEW

File: utils/src/main/java/ua/com/numbers/StringUtils.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package ua.com.numbers;
+
+import static org.apache.commons.lang3.StringUtils.isNumeric;
+
+public class StringUtils {
+    public static void main(String[] args) {
+        System.out.println(StringUtils.isPositiveNumber(args[0]));
+    }
+
+    public static boolean isPositiveNumber(String str) {
```
Comment (sslichnyi) [inline utils/src/main/java/ua/com/numbers/StringUtils.java]: This method should more flexible. 
`isPositiveNumber(String... strs)`

File: utils/src/test/java/ua/com/numbers/StringUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static ua.com.numbers.StringUtils.isPositiveNumber;
+
+class StringUtilsTest {
+    @Test
+    public void testPositiveNumberReturnFalse() {
```
Comment (sslichnyi) [inline utils/src/test/java/ua/com/numbers/StringUtilsTest.java]: Please use ParametrizedTest

Links:
- https://www.baeldung.com/parameterized-tests-junit-5

**Review by sslichnyi — COMMENTED** [review]
Please, change it

File: utils/src/main/java/ua/com/numbers/StringUtils.java (hunk 1 above)
Comment (Fat-Frumos) [reply utils/src/main/java/ua/com/numbers/StringUtils.java]: Made a more flexible method 
boolean isPositiveNumbers(String... args)

File: utils/src/test/java/ua/com/numbers/StringUtilsTest.java (hunk 2 above)
Comment (Fat-Frumos) [reply utils/src/test/java/ua/com/numbers/StringUtilsTest.java]: Added parameterized tests using annotations: @ValueSource, @CsvSource, @MethodSource
Added descriptions for tests

File: Module#01/gradle/multi-project/core/src/main/java/com/epam/esm/core/Utils.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.epam.esm.core;
+
+import ua.com.numbers.StringUtils;
+
+import java.util.Arrays;
+
+public class Utils {
+    public static boolean isAllPositiveNumbers(String... array) {
+        return array != null && array.length != 0 &&
+                Arrays.stream(array).allMatch(StringUtils::isPositiveNumber);
```
Comment (sslichnyi) [inline Module#01/gradle/multi-project/core/src/main/java/com/epam/esm/core/Utils.java]: It is written extremely good with all validation checks.

File: Module#01/gradle/multi-project/core/src/test/java/com/epam/esm/core/UtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Test
+    @DisplayName("Checking for positive numbers and returning true")
+    public void testPositiveNumbersReturnTrue() {
+        Assertions.assertTrue(Utils.isAllPositiveNumbers("123", "312", "321"));
+        Assertions.assertTrue(Utils.isAllPositiveNumbers("1", "2", "3"));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"1", "2", "3"})
```
Comment (sslichnyi) [inline Module#01/gradle/multi-project/core/src/test/java/com/epam/esm/core/UtilsTest.java:30]: Perfect! ParametrizedTest is what I wanted to see here!

File: Module#01/gradle/multi-project/core/build.gradle [hunk 5]
```diff
@@ -0,0 +1,7 @@
+plugins {
+    id 'com.epam.esm.java-library-conventions'
+}
+
+dependencies {
+    implementation files('libs/utils-1.3.5.jar')
```
Comment (sslichnyi) [inline Module#01/gradle/multi-project/core/build.gradle:15]: It is implemented in wrong way. You have to use mavenCentral repository. Please, read more about https://stackoverflow.com/questions/35094655/publish-java-artifact-to-maven-local-with-gradle.

You need to add utils-1.3.5.jar as dependency but not as lib file.

**Comment by sslichnyi** [comment]
please, delete 'git' folder

File: Module#01/gradle/utils/app/src/main/java/com/epam/esm/StringUtils.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package com.epam.esm;
+
+import java.util.Arrays;
+
+import static org.apache.commons.lang3.StringUtils.isNumeric;
+
+public class StringUtils {
+    public static void main(String[] args) {
+        System.out.println(isPositiveNumbers(args));
```
Comment (sslichnyi) [inline Module#01/gradle/utils/app/src/main/java/com/epam/esm/StringUtils.java]: ...Utils class must not contain main() method. StringUtils is more like a library. It is not an application.

File: Module#01/gradle/utils/app/src/main/java/com/epam/esm/StringUtils.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
+    protected static boolean isPositiveNumbers(String... args) {
```
Comment (sslichnyi) [inline Module#01/gradle/utils/app/src/main/java/com/epam/esm/StringUtils.java]: this method should not be protected. it must be public. This method will be used widely in other applications.

File: Module#01/gradle/utils/app/build.gradle [hunk 8]
```diff
@@ -0,0 +1,48 @@
+plugins {
+    id 'java'
+    id 'maven-publish'
+    id 'com.github.johnrengelman.shadow' version '7.0.0'
+}
+
+repositories {
+    mavenLocal()
+    mavenCentral()
```
Comment (sslichnyi) [inline Module#01/gradle/utils/app/build.gradle:9]: It is good. It is exactly what was required.

**Comment by sslichnyi** [comment]
There is no info (commands) in README.md file about steps which programmer should do in order to build and publish your utils jar dependency into local maven repository. Please, provide it.

**Comment by sslichnyi** [comment]
I didn't find any README.md or any other doc, where git task was documented.

**Review by sslichnyi — APPROVED** [review]
Please, fix found issues

File: Module#01/gradle/multi-project/core/build.gradle (hunk 5 above)
Comment (Fat-Frumos) [reply Module#01/gradle/multi-project/core/build.gradle:15]: Added repositories{ maven local, maven central} in build.gradle

File: Module#01/gradle/utils/app/src/main/java/com/epam/esm/StringUtils.java (hunk 6 above)
Comment (Fat-Frumos) [reply Module#01/gradle/utils/app/src/main/java/com/epam/esm/StringUtils.java]: The removed the main method in StringUtils.java

**Pull request merged**